docs/test_development.rst

```diff
@@ -82,64 +80,90 @@ Moreover the unit tests are executed for `each new pull request via Travis
 CI`_.
 
 
-Functional Tests
-====================
+Structure of Functional Tests
+=============================
+
+Setup of *Gluster trusted storage pool(s)* is done prior test execution, and is
+fully automated via `gdeploy config files`_ in `usmqe-setup/gdeploy_config/`_.
+
+No *pytest fixture* or *test case* creates or removes *Gluster trusted storage
+pool(s)* on it's own.
+
+The test cases are implemented as pytest functions, stored in logical chunks in
+python source files (python modules).
+
+Test cases which requires an imported cluster (aka trusted storage pool), uses
+pytest fixture ``imported_cluster``, which:
+
+ * Doesn't create the cluster, but just checks if the cluster is already
+   imported and tries to import it if it's not imported already. If it fails
+   during the import or no suitable cluster is available for import, it
+   raises an errror.
+ * Cluster suitable for import is identified using node defined by
+   ``usm_cluster_member`` parameter in usmqe configuration file.
+ * Returns information about the imported cluster via value of the fixture
+   passed to the test function (``cluster`` object), which includes cluster
+   name, cluster id, volumes in the cluster.
+ * Teardown of this fixture runs cluster unmanage if the cluster was imported
+   during setup phase.
+
+Test casess are tagged by tags:
+
+ * TODO: marker for gluster related tests
+ * TODO: marker for volume type
+ * TODO: marker for happy path tests
+ * TODO: marker for status of gluster profiling
+ * TODO: marker for human readable name
```

Comment:
* marker for working/stable test - currently `mark.testready`
* (maybe?) marker for wip test